tensorflow/lite/c/c_api_experimental.h

```diff
@@ -146,6 +146,43 @@ TFL_CAPI_EXPORT extern void TfLiteInterpreterOptionsSetUseNNAPI(
 TFL_CAPI_EXPORT extern void TfLiteInterpreterOptionsSetEnableDelegateFallback(
     TfLiteInterpreterOptions* options, bool enable);
 
+// Set if buffer handle output is allowed.
+//
+/// When using hardware delegation, Interpreter will make the data of output
+/// tensors available in `tensor->data` by default. If the application can
+/// consume the buffer handle directly (e.g. reading output from OpenGL
+/// texture), it can set this flag to false, so Interpreter won't copy the
+/// data from buffer handle to CPU memory. WARNING: This is an experimental
+/// API and subject to change.
+TFL_CAPI_EXPORT extern void TfLiteSetAllowBufferHandleOutput(
+    const TfLiteInterpreter* interpreter, bool allow_buffer_handle_output);
+
+/// Allow a delegate to look at the graph and modify the graph to handle
+/// parts of the graph themselves. After this is called, the graph may
+/// contain new nodes that replace 1 more nodes.
+/// 'delegate' must outlive the interpreter.
+/// Returns one of the following three status codes:
+/// 1. kTfLiteOk: Success.
+/// 2. kTfLiteDelegateError: Delegation failed due to an error in the
+/// delegate. The Interpreter has been restored to its pre-delegation state.
+/// NOTE: This undoes all delegates previously applied to the Interpreter.
+/// 3. kTfLiteError: Unexpected/runtime failure.
+/// WARNING: This is an experimental API and subject to change.
+TFL_CAPI_EXPORT extern TfLiteStatus TfLiteInterpreterModifyGraphWithDelegate(
```

Comment:
Why is this needed? Does `TfLiteInterpreterOptionsAddDelegate` from `c_api.h` not work? (I want to minimize the new things we add to support this)

Reply:
@srjoglekar246 
I exposed this method for `gl_delegate.h`. `TfLiteGpuDelegateBindBufferToTensor` needs to be called before `TfLiteInterpreterModifyGraphWithDelegate`.

https://github.com/tensorflow/tensorflow/blob/b1bd64aacd5e7a6a918cd8b8557dc9ce23e011bc/tensorflow/lite/delegates/gpu/gl_delegate.h#L115-L121

The order of process on gl_delegate:
 1. Make an interpreter to get model input/output's shape
 2. Create OpenGL SSBO and bind it with the GPU delegate
 3. Call `TfLiteInterpreterModifyGraphWithDelegate`

Reply:
Aah...you are right. Maybe add a note in the docstrings that users should prefer to use `TfLiteInterpreterOptionsAddDelegate` unless absolutely required (like for this use-case)

Reply:
Added a note in 8f99d6b85fe3e198db8ab60315a3db5aff68d8c7